streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamThreadTest.java

```diff
@@ -589,11 +604,51 @@ public void shouldCommitAfterTheCommitInterval() {
         EasyMock.verify(taskManager);
     }
 
+    @Test
+    public void shouldCommitAndStartNewTransactionAfterTheCommitIntervalWithEosBetaEnabled() {
```

Comment:
req: we should also have test for runOnce() user requested commit change

Reply:
Ack.